hack/build-and-push.sh

```diff
@@ -133,95 +180,324 @@ if [ "$SKIP_BUILD" == "" ]; then
-        echo "$output"
-        task_bundle_with_digest="${output##*$'\n'}"
+    (
+        cd "${migration_file%/*}"
+        retry oras attach \
+            --registry-config "$AUTH_JSON" \
+            --artifact-type "$ARTIFACT_TYPE_TEXT_XSHELLSCRIPT" \
+            --distribution-spec v1.1-referrers-tag \
+            --annotation "$ANNOTATION_TASK_MIGRATION=true" \
+            "$task_bundle" "${migration_file##*/}"
+    )
 
-        # copy task to new tag pointing to commit where the file was changed lastly, so that image persists
-        # even when original tag is updated
-        skopeo copy "docker://${task_bundle}" "docker://${task_bundle}-${task_file_sha}"
-    fi
-    # version placeholder is removed naturally by the substitution.
-    real_task_name=$(yq e '.metadata.name' "$prepared_task_file")
-    sub_expr_1="
-        (.spec.tasks[].taskRef | select(.name == \"${real_task_name}\" and .version == \"${task_version}\" ))
-        |= {\"resolver\": \"bundles\", \"params\": [ { \"name\": \"name\", \"value\": \"${real_task_name}\" } , { \"name\": \"bundle\", \"value\": \"${task_bundle_with_digest}\" }, { \"name\": \"kind\", \"value\": \"task\" }] }
-    "
-    sub_expr_2="
-        (.spec.finally[].taskRef | select(.name == \"${real_task_name}\" and .version == \"${task_version}\" ))
-        |= {\"resolver\": \"bundles\", \"params\": [ { \"name\": \"name\", \"value\": \"${real_task_name}\" } , { \"name\": \"bundle\", \"value\": \"${task_bundle_with_digest}\" },{ \"name\": \"kind\", \"value\": \"task\" }] }
-    "
-    for filename in "$generated_pipelines_dir"/*.yaml "$core_services_pipelines_dir"/*.yaml
+    echo
+    echo "Attached migration file ${migration_file} to ${task_bundle}"
+
+    return 0
+}
+
+build_push_tasks() {
+    local build_data
+    local task_bundle_with_digest
+    local migration_file
+    local prepared_task_file
+    local task_file_sha
+    local task_bundle
+
+    find task/*/* -maxdepth 0 -type d | awk -F '/' '{ print $0, $2, $3 }' | \
+    while read -r task_dir task_name task_version
     do
-        yq e "$sub_expr_1" -i "${filename}"
-        yq e "$sub_expr_2" -i "${filename}"
+        if [ -n "$TEST_TASKS" ] && echo "$TEST_TASKS" | grep -v "$task_name" 2>/dev/null; then
+            continue
+        fi
+
+        if should_skip_repo "$QUAY_NAMESPACE" "task-${task_name}"; then
+            echo "NOTE: not pushing task-$task_name:$task_version to $QUAY_NAMESPACE; the repo does not exist and $QUAY_NAMESPACE is deprecated"
+            continue
+        fi
+
+        if is_normal_task "$task_dir" "$task_name";  then
+            build_data=$(generate_normal_task_build_data "$task_dir" "$task_name" "$task_version")
+        elif is_kustomized_task "$task_dir" "$task_name";  then
+            build_data=$(generate_kustomized_task_build_data "$task_dir" "$task_name" "$task_version")
+        else
+            continue
```

Comment:
We should generally never reach the `continue`, right? Maybe echo a warning if this happens? (It would likely mean we're accidentally skipping some task)